File: Chapter 05/5.6 Crypto Tracker/Before/AttemptedTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CryptoTracker.Before.Tests
{
    [TestClass]
    public class AttemptedTests
    {
        [TestMethod]
        public void CalculateTotalValue_UsingLiveApi_IsUntestable()
        {
            // Arrange
            // We create the manager, but it creates its own hardcoded HTTP client inside!
            var manager = new PortfolioManager();

            // Act
            // We want to test the value of 1 BTC.
            var value = manager.CalculateTotalValue(1m);

            // ASSERT
            // Problem: What is the price of Bitcoin right now? 
            // Is it $50,000? $60,000? $20,000?
            // We cannot write a reliable assertion because the data changes every second!
            
            // This test is FLAKY. It will almost certainly FAIL because the live 
            // price is rarely exactly 50,000.
            // Furthermore, it will completely CRASH (throw an exception) if the 
            // test runner doesn't have an active internet connection.
            Assert.AreEqual(50_000m, value);
        }
    }
}

File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_6_crypto_tracker/before/AttemptedTest.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.before;

import com.grokkingsoftwarearchitecture.chapter05.shared.LogManager;

/**
 * A demonstration of why Tight Coupling ruins testability.
 */
public class AttemptedTest {
    public static void run() {
        LogManager.info(AttemptedTest.class, "\n--- ATTEMPTING TO TEST (BEFORE) ---");
        
        PortfolioManager manager = new PortfolioManager();

        LogManager.info(AttemptedTest.class, "Test Action: Calculating value of 1 BTC...");
        
        try {
            double value = manager.calculateTotalValue(1.0);
            
            // ASSERT
            // We cannot assert equality because the price changes constantly.
            LogManager.info(AttemptedTest.class, "Result: {0}", value);
            LogManager.info(AttemptedTest.class, "FAIL: This test is FLAKY. We cannot assert a fixed price.");
        } catch (Exception e) {
            LogManager.info(AttemptedTest.class, "CRASH: Test failed completely. No internet connection.");
        }
    }
}